lib/validation.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+function validateOpts (opts) {
+  const { all, policy, ttl, cacheSize, onHit, onMiss, onSkip } = opts
+
+  if (all && typeof all !== 'boolean') {
```

Comment:
Why the ajv compilation has been removed?

As example, Fastify has an offline script to build a validation function based on the schema:
https://github.com/fastify/fastify/blob/main/build/build-validation.js

To check the `'policy and all options are exclusive'` it is possible to use the [`oneOf`](https://ajv.js.org/json-schema.html#oneof) keyword to match exactly one scheme, not both.

The function check seems doable with a [custom keyword](https://stackoverflow.com/questions/60196587/ajv-check-property-is-a-function)

Reply:
let's not rework this again please, it's good as it is, we don't need ajv